src/Socket.c

```diff
@@ -425,15 +425,18 @@ int Socket_writev(int socket, iobuf* iovecs, int count, unsigned long* bytes)
 	}
 #else
 	*bytes = 0L;
+        assert(count > 0);
 	rc = writev(socket, iovecs, count);
-	if (rc == SOCKET_ERROR)
+	if (rc < 0)
 	{
 		int err = Socket_error("writev - putdatas", socket);
 		if (err == EWOULDBLOCK || err == EAGAIN)
 			rc = TCPSOCKET_INTERRUPTED;
```

Comment:
In this line, the return value is set to -22.

Reply:
writev returns -1 on error, so this shouldn't change anything, right?

Reply:
In line 433 the return value is changed to -22 (TCPSOCKET_INTERRUPTED), and this value is not handled correctly by the caller!